src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -1000,15 +1000,14 @@ public static Task ReplayWarningsAndErrorsAsync(IEnumerable<IAssetsLogMessage> m
         {
             foreach (var framework in frameworks)
             {
-                if (centralPackageVersions.ContainsKey(framework))
+                if (!centralPackageVersions.TryGetValue(framework, out Dictionary<string, CentralPackageVersion> versions))
```

Comment:
The title of this PR made me think it's going against the decisions of https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Build.Tasks/CheckForDuplicateNuGetItemsTask.cs which follows the `compile` item logic where the first one is preferred instead of the last one (as it's what the project-system does). 

This is an edge case on the commandline only right? You can't hit this in VS? 

Can we do first wins strategy to match the rest of the duplicate checking? Or does it not matter at all here.

Reply:
Yeah sorry its confusing, maybe I should update the title?  The duplicates aren't declared in the project and therefore aren't caught by the `CheckForDuplicateNuGetItemsTask`.  The duplicates are actually added by the fact that the same project is processed more than once so we end up with multiple instances in the restore graph items.  Yes this only applies to command-line based restores since they are the only ones affected by this edge case.

> Can we do first wins strategy to match the rest of the duplicate checking? Or does it not matter at all here.

I'm fine with whichever, I'm not sure if it matters since the duplicates are technically added by NuGet itself.  I'm fine either way but would certainly lean towards consistency.

Reply:
I think updating the title would be great. 
This is definitely a weird issue. 

I think consistency would be great, but sounds like the answer to `Or does it not matter at all here` is that it does not actually matter what we choose.

Reply:
I guess we'll stick with this implementation then?

Reply:
I'm fine with whatever. 

If anyone else has a strong opinion one way or the other then we can go with that.

Reply:
I'll go with this for now and add an explicit test for "last one wins"